src/Checkpointer.jl

```diff
@@ -35,8 +46,7 @@ function checkpoint_model_state(
     day = floor(Int, t / (60 * 60 * 24))
     sec = floor(Int, t % (60 * 60 * 24))
     @info "Saving checkpoint " * Interfacer.name(sim) * " model state to HDF5 on day $day second $sec"
-    mkpath(joinpath(output_dir, "checkpoint"))
-    output_file = joinpath(output_dir, "checkpoint", "checkpoint_" * Interfacer.name(sim) * "_$t.hdf5")
+    output_file = joinpath(output_dir, "checkpoint_" * Interfacer.name(sim) * "_$t.hdf5")
     checkpoint_writer = CC.InputOutput.HDF5Writer(output_file, comms_ctx)
```

Comment:
Why do you use `HDF5Writer` here but `jldsave` in `checkpoint_model_cache` below?

Reply:
I'm also unsure of why we don't need to `adapt` here. Is it because the structure of Y is simpler than p?